### PR TITLE
Publish armeria-shaded-for-testing without -all classifier

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

**Review by anuraaga — COMMENTED** [review]
@trask is doing the opposite in another PR :)

**Comment by mateuszrzeszutek** [comment]
> @trask is doing the opposite in another PR :)

😂 
[This one](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3471), isn't it? I guess that our shaded armeria should get the same treatment then.

**Comment by trask** [comment]
The problem is that my PR doesn't work. And this PR causes `armeria-shaded-for-testing` to be published with `<packaging>pom</packaging>` which is what that PR is trying to solve. ♻️

😭

**Comment by mateuszrzeszutek** [comment]
Changed to include `armeria-shaded-for-testing` in `testing-common`, using the method from https://github.com/open-telemetry/opentelemetry-java/pull/3371

File: testing-common/build.gradle.kts [hunk 1]
```diff
@@ -6,6 +6,13 @@ plugins {
 description = "OpenTelemetry Javaagent testing commons"
 group = "io.opentelemetry.javaagent"
 
+sourceSets {
+  main {
+    val traceShadedDeps = project(":testing:armeria-shaded-for-testing")
+    output.dir(traceShadedDeps.file("build/extracted/shadow"), "builtBy" to ":testing:armeria-shaded-for-testing:extractShadowJar")
```
Comment (trask) [inline testing-common/build.gradle.kts:12]:
```suggestion
    val armeriaShadedDeps = project(":testing:armeria-shaded-for-testing")
    output.dir(armeriaShadedDeps .file("build/extracted/shadow"), "builtBy" to ":testing:armeria-shaded-for-testing:extractShadowJar")
```

**Review by trask — APPROVED** [review]
Can this:

```
api(project(path = ":testing:armeria-shaded-for-testing", configuration = "shadow"))
```

be changed to this now?

```
compileOnly(project(":testing:armeria-shaded-for-testing"))
```

File: testing/armeria-shaded-for-testing/build.gradle.kts [hunk 2]
```diff
@@ -28,4 +25,14 @@ tasks {
     relocate("META-INF/native/netty", "META-INF/native/io_opentelemetry_testing_internal_netty")
     mergeServiceFiles()
   }
-}
\ No newline at end of file
+
+  val extractShadowJar by registering(Copy::class) {
```
Comment (iNikem) [inline testing/armeria-shaded-for-testing/build.gradle.kts:29]: I now propose to stop and take a step back. Extracting shadow jar that we have just built seems like we are doing something very-very wrong.

**Comment by mateuszrzeszutek** [comment]
> Can this:
> 
> ```
> api(project(path = ":testing:armeria-shaded-for-testing", configuration = "shadow"))
> ```
> 
> be changed to this now?
> 
> ```
> compileOnly(project(":testing:armeria-shaded-for-testing"))
> ```

It still needs to refer to the `shadow` configuration to see the relocated packages, but it sure can be `compileOnly`

**Review by anuraaga — APPROVED** [review]

**Pull request merged**